addons/pos_loyalty/static/src/overrides/components/payment_screen/payment_screen.js

```diff
@@ -97,6 +97,7 @@ patch(PaymentScreen.prototype, {
             ) {
                 agg[pe.coupon_id].partner_id = partner.id;
             }
+            agg[pe.coupon_id].date_to = program.date_to;
```

Comment:
Should not set date_to for program type loyalty see 17.4 https://github.com/odoo/odoo/pull/195202/files for reference
```suggestion
            if (program.program_type != "loyalty") {
                agg[pe.coupon_id].date_to = program.date_to;
            }
```